packages/platform-sdk-btc/src/services/wallet-discovery.ts

```diff
@@ -0,0 +1,38 @@
+import { Coins, Contracts } from "@arkecosystem/platform-sdk";
+import Bitcoin from "bitcore-lib";
+
+import { bip44, bip49, bip84 } from "./identity/utils";
+
+export class WalletDiscovery implements Contracts.WalletDiscoveryService {
+	readonly #network: Record<string, any>;
+
+	public constructor(config: Coins.Config) {
+		this.#network = Bitcoin.Networks[config.get<Coins.NetworkManifest>("network").id.split(".")[1]];
+	}
+
+	public async fromMnemonic(mnemonic: string): Promise<Contracts.AddressDiscoveryDataTransferObject[]> {
+		const [nonSegwit, segwit, nativeSegwit] = await Promise.all([
```

Comment:
for consistency

```suggestion
		const [legacy, segwit, nativeSegwit] = await Promise.all([
```

Reply:
Legacy is another schema that is not yet supported so it will be misleading to name it this.